docs/exposing-binding-data/olm-descriptors.md

```diff
@@ -0,0 +1,184 @@
+  - service.binding
+```
+
+The X-Descriptor has `urn:alm:descriptor:io.kubernetes:ConfigMap` entry to
+indicate the path is pointing to a ConfigMap resource.
+
+## Extract an entry from a ConfigMap/Secret into the binding Secret
+
+When you want to extract only a particular entry from a Secret or ConfigMap, the X-Descriptor can
+update `service.binding` line with a name and `sourceKey`.  Here is an example:
+
+```
+- path: data.dbConfiguration
+  x-descriptors:
+  - urn:alm:descriptor:io.kubernetes:ConfigMap
+  - service.binding:my_certificate:sourceKey=certificate
+```
+
+In the above example, `sourceKey` points to the name of the key in the Secret
+resource, which is `certificate`.  And `my_certificate` is the name of the
+binding key that's going to be injected.
+
+## Extract a resource definition value into the binding Secret
+
+When values required for binding to the backing service are available as
+attributes of its resources, it's possible to annotate these values (which will
+identify them as "binding metadata"), using an X-Descriptors with name.  Here is
+an example:
+
+```
+- path: data.connectionURL
+  x-descriptors:
+  - service.binding:uri
+```
+
+In the above example, the `connectionURL` attribute points to the required
+value.  It will be injected as `uri`.
+
+## Extract the entries of a collection into the binding Secret selecting the key and value from each entry
+
+Consider this example from a backig service resource:
+
+```
+status:
+  connections:
+    - type: primary
+      url: primary.example.com
+    - type: secondary
+      url: secondary.example.com
+    - type: '404'
+      url: black-hole.example.com
+```
+
+If you want to inject all those values with key as `primary`, `secondary` etc.,
+you can an X-Descriptors like this:
+
+```
+- path: bootstrap
+  x-descriptors:
+  - service.binding:endpoints:elementType=sliceOfMaps:sourceKey=type:sourceValue=url
+```
+
+## Extract the items of a collection into the binding Secret with one key per item
+
+Consider this example from a backig service resource:
+
+```
+spec:
+    tags:
+      - knowledge
+      - is
+      - power
+```
+
+If you want to inject all those values with key as `prefix_0`, `prefix_1` etc.,
+you can an X-Descriptors like this.  The default prefix is the name of the
+resource kind:
+
+```
+- path: spec.tags
+  x-descriptors:
+  - service.binding:tags:elementType=sliceOfStrings
+```
+
+## Extract the values of collection entries into the binding Secret with one key per entry value
+
+Consider this example:
+
+```
+spec:
+    connections:
+      - type: primary
+        url: primary.example.com
+      - type: secondary
+        url: secondary.example.com
+      - type: '404'
+        url: black-hole.example.com
+```
+
+If the structure is like this, and you want to inject all those values with key
+as `prefix_0`, `prefix_1` etc., you can an X-Descriptors like this.  The default
+prefix is the name of the resource kind:
+
```

Comment:
Insert the content:

- "### Example: Configuration for the required values of collection entries"

Reply:
Done